test/e2e/framework/cluster.go

```diff
@@ -0,0 +1,352 @@
+// Copyright (C) 2024 ScyllaDB
+
+package framework
+
+import (
+	"context"
+	"crypto/sha512"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	scyllaclientset "github.com/scylladb/scylla-operator/pkg/client/scylla/clientset/versioned"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/discovery"
+	cacheddiscovery "k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+type Cluster interface {
+	Namespace() string
```

Comment:
I think this belong to the framework only (and is implemented by the cluster 0, to mimic the it block. After that when you want a namespace in a different cluster, you ask for it. But we shouldn't assume you need a namespace in all the clusters, or that it will be present by default. Only the framework gets the default namespace guaranteed (unless you disable it but I don't think that implemented today).

Reply:
https://github.com/scylladb/scylla-operator/pull/1881#discussion_r1611217425